### PR TITLE
Enable Google One Tap by tenant

### PR DESCRIPTION
**Pull request by indeewari**
### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/core/models/config.ts [hunk 1]
```diff
@@ -191,6 +191,10 @@ export interface UIConfigInterface extends CommonUIConfigInterface<FeatureConfig
      * `REMOTE` will fetch templates from the template management REST API.
      */
     applicationTemplateLoadingStrategy?: ApplicationTemplateLoadingStrategies;
+    /**
+     * Configuration to enable Google One Tap for specific tenants.
+    */
```
Comment (JayaShakthi97) [inline apps/console/src/features/core/models/config.ts:196]:
```suggestion
     */
```

**Comment by somindatommy** [comment]
@indeewari In here we are trying to retrieve the config with the tenant list right? Instead, can we have a method to check whether the tenant is skipped or not? A method with boolean return type?

File: features/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2 [hunk 2]
```diff
@@ -181,9 +181,13 @@
         "identityProviderTemplateLoadingStrategy": "{{ console.ui.identity_provider_template_loading_strategy }}",
         {% endif %}
         "appLogoPath": "{{ console.ui.app_logo_path }}",
-        {% if console.enable_google_one_tap_for_super_tenant_only is defined %}
-        "isGOTEnabledForSuperTenantOnly": "{{ console.enable_google_one_tap_for_super_tenant_only }}",
-        {% endif %}
+        "googleOneTapEnabledTenants": [
+            {% if console.google_one_tap_enabled_tenants is defined %}
```
[the remark this answers is not in the document]
Comment (indeewari) [reply features/org.wso2.identity.apps.console.server.feature/resources/deployment.config.json.j2]: This is fixed c331826f2c27a2b48167c75c536bc3b90f258c6d

File: apps/console/src/features/core/models/config.ts (hunk 1 above)
Comment (indeewari) [reply apps/console/src/features/core/models/config.ts]: This is fixed c331826f2c27a2b48167c75c536bc3b90f258c6d

**Review by thivi — DISMISSED** [review]

File: apps/console/src/features/identity-providers/components/forms/authenticators/google-authenticator-form.tsx [hunk 3]
```diff
@@ -232,22 +232,22 @@ export const GoogleAuthenticatorForm: FunctionComponent<GoogleAuthenticatorFormP
...
-        return !isGOTEnabledForSuperTenantOnly || isSuperTenantLogin();
+        if (googleOneTapEnabledTenants?.length > 0) {
+            return googleOneTapEnabledTenants.includes(AppConstants.getTenant());
+        }
+
+        /**
+         * To enable Google One Tap for all tenants
+         * if this configuration is not defined or empty.
+         */
+
```
Comment (NipuniBhagya) [inline apps/console/src/features/identity-providers/components/forms/authenticators/google-authenticator-form.tsx:249]:
```suggestion
```

**Review by NipuniBhagya — DISMISSED** [review]

File: apps/console/src/features/identity-providers/components/forms/authenticators/google-authenticator-form.tsx (hunk 3 above)
Comment (indeewari) [reply apps/console/src/features/identity-providers/components/forms/authenticators/google-authenticator-form.tsx]: This is fixed 90114ebb54a8820fdf7882291874775c7d6becc0

**Review by JayaShakthi97 — APPROVED** [review]

**Review by somindatommy — APPROVED** [review]

**Comment by indeewari** [comment]
> @indeewari In here we are trying to retrieve the config with the tenant list right? Instead, can we have a method to check whether the tenant is skipped or not? A method with boolean return type?

This will be incorporated as an improvement later.

**Pull request merged**